### DISCUSSION TITLE
Unable to install PETsc

### DISCUSSION THREAD

**Discussion by CarsonKipp** [firedrakeproject/firedrake#3273] [Firedrake support] [answered]
I saw the earlier post where a user was receiving the same error message that I'm receiving below. However, I'm using an Ubuntu distribution on my windows machine and I am also receiving the following error when reaching the PETsc portion of the "python3 firedrake-install" command. I felt this was worth mentioning as the current suspect in the other discussion is their homebrew installation, however the issue persists without using homebrew:

Traceback (most recent call last):
  File "firedrake-install", line 1907, in <module>
    install("petsc/")
  File "firedrake-install", line 1051, in install
    build_and_install_petsc()
  File "firedrake-install", line 1163, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "firedrake-install", line 664, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/ckipp/firedrake/bin/python', './configure', 'PETSC_DIR=/home/ckipp/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-chaco', '--with-fortran-bindings=0', '--download-superlu_dist', '--download-hdf5', '--with-c2html=0', '--download-suitesparse', '--download-netcdf', '--with-zlib', '--with-shared-libraries=1', '--download-cmake', '--download-bison', '--download-hwloc', '--download-scalapack', '--download-hypre', '--download-metis', '--with-debugging=0', '--download-pastix', '--download-mumps', '--download-ptscotch', '--download-mpich', '--download-pnetcdf']' returned non-zero exit status 1.

**Comment by connorjward** [comment]
Please share the `firedrake-install.log` and `firedrake/src/petsc/configure.log` files.

**Reply by CarsonKipp** [reply]
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/13492385/firedrake-install.log)

I see the configure.log file in the petsc directory you mentioned however when I try to attach it, it says "the system cannot find the specified path"

**Reply by connorjward** [reply, marked as answer]
That's OK. The installation log you shared is sufficient. It states

`Cannot build Bison. It requires either "flex" or "lex" in PATH. Please install flex and
  retry.`

flex is listed as one of our system requirements [here](https://www.firedrakeproject.org/download.html). I think a simple `apt install flex bison` will fix things for you though it may be good to also install the other dependencies.

Since you're on Ubuntu and we claim to install these dependencies for you this might be a bug. But regardless this should hopefully fix things for you.
